packages/compile-solidity-tests/test/settings.ts

```diff
@@ -0,0 +1,43 @@
+import { assert } from "chai";
+import Config from "@truffle/config";
+import { Compile } from "@truffle/compile-solidity";
+import { describe, it } from "mocha";
+import path from "path";
+import { Environment } from "@truffle/environment";
+
+describe("Compile with disableDefaults flag", () => {
+  it("should not add the optimizer when missing if disableDefaults is true", async () => {
+    // Load configuration from the fixture file
+    const config = Config.detect(
+      { workingDirectory: path.join(__dirname, "fixture/default-box") },
+      "./truffle-config.js"
+    );
+
+    // Environment.detect() so we get all the other config goodies
+    Environment.detect(config);
+
+    //compile when optimizer is set to false
+    const compilationWithOptimizer = await Compile.all(config);
+
+    //get the bytecode for the `optimizer: false` case
+    const bytecodeOptimizer =
```

Comment:
Naming this something like `bytecodeUsingDefaults` might make the flow of the test here more clear

Reply:
Omg thank you for saving me from my "unable to properly name things" weakness. 😄 I'll update these.

Reply:
Although, "using defaults" seems too broad? We really are only testing the optimizer. I went with `bytecodeUsingDefaultOptimizer` and `bytecodeWithoutDefaultOptimizer` but if you feel strongly I'll change it to what you originally suggested!